src/main.tsx

```diff
@@ -51,22 +60,17 @@ export default class ChessStudyPlugin extends Plugin {
 			editorCallback: (editor: Editor) => {
 				const cursorPosition = editor.getCursor();
 
-				const onSubmit = async (pgn_or_fen: string) => {
+				const onSubmit = async (chessString: ChessString | undefined) => {
 					try {
-						let pgn = '', fen = '';
-						if (pgn_or_fen) {
-							if (pgn_or_fen.includes('/')) {
-								fen = pgn_or_fen.trim();
-							} else {
-								pgn = pgn_or_fen.trim();
-							}
-						}
+						const chessStringTrimmed = chessString?.trim() ?? '';
 
-						const chess = (fen) ? new Chess(fen) : new Chess();
+						const isFen = chessStringTrimmed.includes('/');
 
-						if (pgn) {
+						const chess = isFen ? new Chess(chessStringTrimmed) : new Chess();
```

Comment:
You can actually pass a FEN string or a PGN string into the `Chess` constructor. It accepts both.

Reply:
I had to use the `strict: false` flag to get it to work properly for some cases, but we can try to not use that flag.

Reply:
I actually ran into the same issue. I think you're right. It's more consistent to use the loadPgn.